File: langgraph/result/qwen/7b-eq2js/55.js
function calculateFibonacci(n) {
    if (n === 0) {
        return 0;
    } else if (n === 1) {
        return 1;
    }

    let previous = 0, current = 1, index = 2;
    while (index <= n) {
        [previous, current] = [current, previous + current];
        index++;
    }

    return current;
}

File: langgraph/data/semantic_equiv/55.py
def calculate_fibonacci(index: int):
    if index == 0:
        return 0
    elif index == 1:
        return 1
    else:
        previous, current = 0, 1
        for _ in range(2, index + 1):
            previous, current = current, previous + current
        return current

# method name: fib -> calculate_fibonacci
# parameter name: n -> index
# replaced recursion with an iterative approach using a for loop
# introduced variables: previous, current to store Fibonacci numbers
# used tuple unpacking to update previous and current values